pkg/systemd/services/services.jsx

```diff
@@ -351,42 +392,30 @@ class ServicesPageBody extends React.Component {
     }
 
-    /* When the page is running in the background fetch only information about failed units
-     * in order to update the 'Page Status'. The whole listUnits is very expensive.
-     * We still need to maintain the 'unit_by_path' state object so that if we receive
-     * some signal we can normally parse it and update only the affected unit state
-     * instead of calling ListUnitsFiltered API call for every received signal which
-     * might have changed the failed units array
-     */
+    /* When the page is running in the background, fetch only information about failed units
+     * in order to update the 'Page Status'. */
     listFailedUnits() {
-        return systemd_client[this.props.owner].call(SD_OBJ, SD_MANAGER, "ListUnitsFiltered", [["failed"]])
+        return systemd_client[this.props.owner].call(s_bus.O_MANAGER, s_bus.I_MANAGER, "ListUnitsFiltered", [["failed"]])
                 .then(([failed]) => {
-                    failed.forEach(result => {
-                        const path = result[6];
-                        const unit_id = result[0];
-
-                        if (!this.isUnitHandled(unit_id))
+                    const units = {};
+                    const unit_by_path = {};
+                    failed.forEach(([
+                        Id, Description, LoadState, ActiveState, _substate, _followUnit, ObjectPath,
+                        _is_job_queued, _job_type, _job_path
+                    ]) => {
+                        if (!this.isUnitHandled(Id))
                             return;
 
-                        // Ignore units which 'not-found' LoadState
-                        if (result[2] == 'not-found')
-                            return;
+                        unit_by_path[ObjectPath] = Id;
 
-                        if (!this.seenPaths.has(path))
-                            this.seenPaths.add(path);
-
-                        this.updateProperties(
-                            {
-                                Id: cockpit.variant("s", unit_id),
-                                Description: cockpit.variant("s", result[1]),
-                                LoadState: cockpit.variant("s", result[2]),
-                                ActiveState: cockpit.variant("s", result[3]),
-                                SubState: cockpit.variant("s", result[4]),
-                            }, path
-                        );
+                        units[Id] = { Id, Description, LoadState, ActiveState, ObjectPath };
                     });
-                    this.processFailedUnits();
-                }, ex => console.warn('ListUnitsFiltered failed: ', ex.toString()));
+
+                    this.unit_by_path = unit_by_path;
+                    this.setState({ units, isFullyLoaded: false });
+                    this.processFailedUnits(units);
+                })
+                .catch(ex => console.warn('ListUnitsFiltered failed: ', ex.toString()));
```

Comment:
This added line is not executed by any test.  [Details](https://cockpit-logs.us-east-1.linodeobjects.com/pull-18422-20230303-165501-04bdc094-fedora-37-devel//Coverage/lcov/github-pr.diff.gcov.html)